### PR TITLE
Add home page compatibility for Twenty Twenty-One theme

### LINKED ISSUE #28308: Twenty Twenty One home page
Please ensure the 2021 home page (similar to what storefront creates) looks good:

- [ ] wide screen layout
- [ ] middle screen layout (e.g. ipad)
- [ ] narrow screen layout (e.g. iphone)
- [ ] test with nonsquare product images (narrow and horizontally wider)
- [ ] Chrome
- [ ] Safari
- [ ] Firefox

Make sure the transitions between the layout widths look sensible (e.g. if the menu is still in narrow layout mode but the rest of the content in wide layout mode, it might be suboptimal, but please use your best judgment).

### PR DESCRIPTION
**Pull request by rodrigoprimo**
### Changes proposed in this Pull Request:

I'm starting this PR to make the changes necessary so that the home page is displayed nicely when using the Twenty Twenty-One theme.

So far, I have adjusted the size of the background image in the cover block, the `<h2>` elements used as section titles, and started to adjust the shortcode that displays a list of categories.

To finish this PR, we have to make some remaining changes to the list of categories, work on the list of products, and check the Figma design to see if something else is missing. We should probably coordinate with @nerrad's regarding the list of products that are displayed on the home page as it uses a product block.

I don't write CSS very often so there is a high chance that there are better ways to implement the changes proposed in this PR. Any feedback is welcome. cc @claudiosanches as you have more experience with CSS.

As we discussed, I'm opening this PR against `add/2021-compat` instead of `master`.

Closes #28308

### How to test the changes in this Pull Request:

Pending

### Changelog entry

Pending

### CODE REVIEW

**Comment by nerrad** [comment]
Anything related to blocks should be done as a part of https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/3433 which should land in an eventual WooCommerce Blocks 3.8.1 patch release and brought over as a package bump for WooCommerce Core 4.8 (assuming WC 4.8 is the release targeting WP 5.6 and Twenty Twenty One compat?)

File: assets/css/twenty-twenty-one.scss
Comment (jonathansadowski) [inline assets/css/twenty-twenty-one.scss:2150]: Calling out my own change here while I'm noticing it. I switched what was `margin` to `padding` so that the `border-box` calculations would be correct, but this appears to be affecting other pages.

Can probably switch this back to margin and account for it in the width of `.columns-n li.product` using `calc`.

File: assets/css/twenty-twenty-one.scss
Comment (jonathansadowski) [inline assets/css/twenty-twenty-one.scss:2482]: I'm wondering if we should modify styles outside of a `woocommerce` class... in theory, this would potentially affect the cover for non-WooCommerce homepages.

I also removed this and didn't notice any difference.

**Review by ObliviousHarmony — APPROVED** [review]
LGTM, once we have all of these merged into #28134 we can see if there's any other issues to fix and if anything squashed anything else!

**Pull request merged**
